src/test/java/com/example/solidconnection/university/service/UniversityRecommendServiceTest.java

```diff
@@ -0,0 +1,170 @@
+                ));
+    }
+
+    @Test
+    void 관심_국가_설정한_사용자의_맞춤_추천_대학을_조회한다() {
+        // given
+        SiteUser testUser = createSiteUser();
+        interestedCountyRepository.save(new InterestedCountry(testUser, 덴마크));
+
+        // when
+        UniversityRecommendsResponse response = universityRecommendService.getPersonalRecommends(testUser.getEmail());
+
+        // then
+        assertThat(response.recommendedUniversities())
+                .hasSize(RECOMMEND_UNIVERSITY_NUM)
+                .containsAll(List.of(
+                        UniversityInfoForApplyPreviewResponse.from(서던덴마크대학교_지원_정보),
+                        UniversityInfoForApplyPreviewResponse.from(코펜하겐IT대학_지원_정보)
+                ));
+    }
+
+    @Test
+    void 관심_지역과_국가_모두_설정한_사용자의_맞춤_추천_대학을_조회한다() {
+        // given
+        SiteUser testUser = createSiteUser();
+        interestedRegionRepository.save(new InterestedRegion(testUser, 영미권));
+        interestedCountyRepository.save(new InterestedCountry(testUser, 덴마크));
+
+        // when
+        UniversityRecommendsResponse response = universityRecommendService.getPersonalRecommends(testUser.getEmail());
+
+        // then
+        assertThat(response.recommendedUniversities())
+                .hasSize(RECOMMEND_UNIVERSITY_NUM)
+                .containsExactlyInAnyOrder(
+                        UniversityInfoForApplyPreviewResponse.from(괌대학_A_지원_정보),
+                        UniversityInfoForApplyPreviewResponse.from(괌대학_B_지원_정보),
+                        UniversityInfoForApplyPreviewResponse.from(메모리얼대학_세인트존스_A_지원_정보),
+                        UniversityInfoForApplyPreviewResponse.from(네바다주립대학_라스베이거스_지원_정보),
+                        UniversityInfoForApplyPreviewResponse.from(서던덴마크대학교_지원_정보),
+                        UniversityInfoForApplyPreviewResponse.from(코펜하겐IT대학_지원_정보)
+                );
+    }
+
+    @Test
+    void 관심사_미설정_사용자는_일반_추천_대학을_조회한다() {
+        // given
+        SiteUser testUser = createSiteUser();
+
+        // when
+        UniversityRecommendsResponse response = universityRecommendService.getPersonalRecommends(testUser.getEmail());
+
+        // then
+        assertThat(response.recommendedUniversities())
+                .hasSize(RECOMMEND_UNIVERSITY_NUM)
+                .containsExactlyInAnyOrderElementsOf(
+                        generalRecommendUniversities.getRecommendUniversities().stream()
+                                .map(UniversityInfoForApplyPreviewResponse::from)
+                                .toList()
+                );
+    }
+
+    @Test
+    void 존재하지_않는_사용자_조회시_예외를_반환한다() {
+        // given
+        String invalidEmail = "invalid@email.com";
+
+        // when & then
+        CustomException exception = assertThrows(CustomException.class,
+                () -> universityRecommendService.getPersonalRecommends(invalidEmail));
+
+        assertThat(exception.getMessage()).isEqualTo(USER_NOT_FOUND.getMessage());
```

Comment:
```suggestion
        assertThatCode(() -> universityRecommendService.getPersonalRecommends(invalidEmail))
                .isInstanceOf(CustomException.class)
                .hasMessage(USER_NOT_FOUND.getMessage());
```

이렇게 하면 한줄로도 검증할 수 있어요!

Reply:
아하 새로 알았네요 감사합니다!

Reply:
이거 적용해봤는데 CustomException이 아닌 RuntimeException 이 뜬다고 테스트코드가 실패하더라구요 혹시 이유 알고계신가요?

Reply:
아 getUniversityDetail의 경우 현재 캐싱 AOP가 적용되어 있어서 예외가 바로 CustomException으로 오지 않고 RuntimeException으로 wrapping되어서 오는 거 같습니다

Reply:
1. RuntimeException이 발생하는지 확인
2. root cause를 가져와서
3. 그게 CustomException인지 확인하고
4. 그 예외의 메시지가 기대하는 메시지와 일치하는지 확인

이과정으로 하는 건 괜찮을까요?
```
 @Test
    void 존재하지_않는_대학_상세정보_조회시_예외_응답을_반환한다() {
        // given
        Long invalidUniversityInfoForApplyId = 9999L;

        // when & then
        assertThatExceptionOfType(RuntimeException.class)
                .isThrownBy(() -> universityQueryService.getUniversityDetail(invalidUniversityInfoForApplyId))
                .havingRootCause()
                .isInstanceOf(CustomException.class)
                .withMessage(UNIVERSITY_INFO_FOR_APPLY_NOT_FOUND.getMessage());
    }
```

Reply:
네 적절해보입니다😊